service/src/main/java/org/mskcc/smile/service/impl/AwsS3ServiceImpl.java

```diff
@@ -0,0 +1,221 @@
+            System.out.println("SAML2AWS Login Output: " + line);
+        }
+
+        int exitCode = loginProcess.waitFor();
+        System.out.println("SAML2AWS Login Exit Code: " + exitCode);
+
+        if (exitCode > 0) {
+            throw new RuntimeException("Failed to run: " + StringUtils.join(saml2AwsCmd, " "));
+        }
+    }
+
+    /**
+     * TO-DO: GET THE CREDENTIALS REFRESH WORKING CORRECTLY
+     * See doc related to AWS SDK Identity
+     * - https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/identity/spi/Identity.html
+     *      expiration time = The time after which this identity will no longer be valid.
+     *      if this is empty, an expiration time is not known (but the identity may still
+     *      expire at some time in the future).
+     * @param s3Client
+     * @return
+     */
+    private Boolean sessionIsExpired(S3Client s3Client) {
+        if (s3Client == null) {
+            return Boolean.TRUE;
+        }
+        try {
+            IdentityProvider credentialsProvider
+                    = s3Client.serviceClientConfiguration().credentialsProvider();
+
+            System.out.println("\n\nprinting credentials provider");
+            System.out.println(credentialsProvider);
+            System.out.println("\nprinting credentials provider identity type");
+            System.out.println(credentialsProvider.identityType());
+            System.out.println("\n\n");
+
+            AwsCredentialsIdentity credentialsIdentity =
+                    (AwsCredentialsIdentity) credentialsProvider.resolveIdentity().join();
+            if (credentialsIdentity instanceof AwsSessionCredentials) {
+                System.out.println("Identified credentials identity is temporary meaning "
+                        + "that there is a time-based expiration...");
+
+                AwsSessionCredentials sessionCredentials = (AwsSessionCredentials) credentialsIdentity;
+                if (sessionCredentials.expirationTime().isPresent()) {
+                    Instant expirationTime = sessionCredentials.expirationTime().get();
+                    Instant now = Instant.now();
+                    System.out.println("\n\nexpiration time");
+                    System.out.println(expirationTime);
+                    System.out.println("time now");
+                    System.out.println(now);
+                    System.out.println("\n\n");
+
+                    Duration timeLeft = Duration.between(now, expirationTime);
+                    if (timeLeft.isPositive()) {
+                        return Boolean.FALSE;
+                    } else {
+                        return Boolean.TRUE;
+                    }
+                } else {
+                    System.out.println("Expiration time not available for these credentials even though "
+                            + "credentialsIdentity identified as instanceof 'AwsSessionCredentials'.");
+                    return Boolean.FALSE;
```

Comment:
@n1zea144 Still working this stuff out. I've verified that a session with an expiration is in fact generated but the session credentials that I extract from the `s3Client` is not a type that has an expiration time even though it is considered an instance of `AwsSessionCredentials` which should have a non-empty expiration time